### PR TITLE
Fix Shortlist filtering

### PR DESCRIPTION
**Pull request by pezholio**
For mysociety/fixmystreetforcouncils#146

This adds the JSON response to the controller action, and wraps the problem list in a `js-reports-list` container, so the list can be updated.

### CODE REVIEW

**Comment by pezholio** [comment]
Ah, good catch. I'm sure there'll be a way round this. I'll have a peek 👍

**Comment by pezholio** [comment]
This should be sorted now 👍

File: perllib/FixMyStreet/App/Controller/My.pm [hunk 1]
```diff
@@ -57,6 +57,7 @@ sub planned : Local : Args(0) {
     $c->forward('/reports/stash_report_sort', [ 'shortlist' ]);
     $c->forward('get_problems');
     if ($c->get_param('ajax')) {
+        $c->stash->{shortlist} = (!defined $c->get_param('sort') || $c->get_param('sort') eq 'shortlist');
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/My.pm]: You can just use the stash->{sort_key} here (which has already done param checking for you).

Comment (pezholio) [inline perllib/FixMyStreet/App/Controller/My.pm]: Ah, nice 👍

**Pull request merged**